### PR TITLE
RGRIDT-687: MarkChangesAsSaved behavior with invalid cells;

### PR DESCRIPTION
**Pull request by JoaoFerreira-FrontEnd**
### What was happening
* When the user was calling the MarkChangesAsSaved, we were cleaning the DirtyMarks and the ValidationMarks.

### What was done
* I've changed the API and all the methods including the provider code, so we can choose if we should clean the validation marks or not, using a dynamic flag (boolean).

### Test Steps
1. Build a grid with a SetValidationStatus on the OnCellValueChange;
2. Change the validation status, IsValid value to False;
3. On another button use a new action that uses MarkChangesAsSaved client action;
4. Config the value of ClearInvalids to False;
5. Call this action after changing the value on the previous column;
6. The invalid status mark should continue on the cell;
7. Config the value of ClearInvalids to True;
8. Repeat the test;
9. The invalid status mark shouldn't be displayed;

### Checklist
* [x] tested locally
* [x] documented the code
* [x] clean all warnings and errors of eslint
* [x] requires changes in OutSystems (if so, provide a module with changes)
* [x] requires new sample page in OutSystems (if so, provide a module with changes)

### CODE REVIEW

File: code/src/OSFramework/Interface/IRowMetadata.ts [hunk 1]
```diff
@@ -22,31 +22,64 @@ namespace OSFramework.Interface {
...
-        clearPropertyByRow(row: number, propertyName: string): void;
+        clearPropertyByRowNumber(row: number, propertyName: string): void;
         /**
          * Responsible for retrieving metadata information for a given row and property
          * @param row Index row reference, works only in the current page
          * @param propertyName Metadata property
          */
         // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        getMetadata(row: number, propertyName: string): any;
+        getMetadataByRow(row: any, propertyName: string): any;
```
Comment (rugoncalves) [inline code/src/OSFramework/Interface/IRowMetadata.ts:42]: I would change the name to _getMetadataInRow_. This is a more direct name :)

**Review by rugoncalves — APPROVED** [review]
Minor name reviews if you think will look better. :) 
Congratulations for all this work 🥇

**Review by rugoncalves — APPROVED** [review]

File: code/src/OSFramework/Feature/IDirtyMark.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@ namespace OSFramework.Feature {
     export interface IDirtyMark {
         clear(): void;
         // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        clearPropertyInRow(row: any): void;
```
Comment (tmbp95) [inline code/src/OSFramework/Feature/IDirtyMark.ts:6]: Can you please add  a comment here? We should have comments all over the interfaces to make it easier for the other developers.

**Review by tmbp95 — APPROVED** [review]

**Review by bmartinho — APPROVED** [review]
LGTM

**Pull request merged**